File: rand.scala
#!/usr/local/scala/scala-2.13.4/bin/scala
exec scala "$0" "$@"
!#
object rand{
  def random(m: Int, a: Int, c: Int, x: Int): Int = {
    return (a * x + c) % m;
  }

  def main(args: Array[String]): Unit = {
    if (args.length != 5) {
      println(random(args(0).toInt, args(1).toInt, args(2).toInt, args(3).toInt));
    }
    else{
      println("Error: Not matching argument count.");
    }
  }
}

File: rand.R
#!/usr/bin/Rscript
random <- function(m, a, c, x){
  return((a * x + c) %% m)
}

args <- commandArgs()
if (length(args) == 9) {
  cat(random(strtoi(args[6]), strtoi(args[7]), strtoi(args[8]), strtoi(args[9])))
} else {
  cat("Error: Not matching argument count.")
}
